### PR TITLE
Add attribution for keyboard shortcuts

### PR DESCRIPTION
**Pull request by ben-lu-uw**
Closes https://github.com/Maps4HTML/Web-Map-Custom-Element/issues/267
Add an attribution that opens a dialog element to show keyboard shortcuts. 

Dialog element is not supported in Internet Explorer, not sure if this is a concern or not given that it has been retired.

### CODE REVIEW

**Comment by prushforth** [comment]
> Dialog element is not supported in Internet Explorer, not sure if this is a concern or not given that it has been retired.

Nope, because custom elements aren't supported there either, so this doesn't matter for that antecedent reason.

**Comment by prushforth** [comment]
When I load the dist branch into experiments, I get this:

image

**Comment by ben-lu-uw** [comment]
If you have the extension install and open a page with the map-options hard coded into the html for sake of demo that will happen.

**Comment by prushforth** [comment]
Is the tab stop at the features removed inadvertently with this link? When tab from feature index (focus on map) it lands not on the features but on the attribution control, I think.

**Comment by ben-lu-uw** [comment]
The tab order should be the same as it was originally. Map -> features -> controls -> attributions.

**Comment by prushforth** [comment]
> If you have the extension installed without the corresponding branch and open a page with the map-options hard coded into the html for sake of demo that will happen.

So I do have the extension installed, and when the extension has the corresponding branch installed, I get the english dialog (did you localize the dialog content?  I find [deepl](https://www.deepl.com/translator) to be a great resource for the programmer who does it all).  The [restaurants page](https://maps4html.org/experiments/screenreader/restaurants/) doesn't have the map-options hard-coded, so it means that that's what a user who doesn't have the extension installed well get.

> With feature indexing on, it was originally decided that the features would not be part of the tab order until arrow key feature navigation got implemented. 

I forgot that it's not implemented yet, awaiting review 🤦‍♂️ ! 😄

**Comment by ben-lu-uw** [comment]
I misspoke about the undefined part. That will happen if you have an old version of the extension. If the extension is not installed it uses the defaults set up in options.js.

**Comment by prushforth** [comment]
I sent a PR with french and swedish localization (hopefully @Malvoz will vet my swedish).  One issue I notice is when I set chrome to display in french, the "Keyboard Shortcuts" attribution link is not localized:

image
image

File: src/mapml/control/AttributionControl.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    map._attributionControl = map._map.attributionControl.setPrefix('<button onclick="this.closest(\'.leaflet-container\').querySelector(\'.shortcuts-dialog\').showModal()" class="shortcuts-button mapml-button">Keyboard shortcuts</button> | <a href="https://www.w3.org/community/maps4html/" title="W3C Maps for HTML Community Group">Maps4HTML</a> | <img src="data:image/svg+xml;base64,PHN2ZyBhcmlhLWhpZGRlbj0idHJ1ZSIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB3aWR0aD0iMTIiIGhlaWdodD0iOCI+PHBhdGggZmlsbD0iIzRDN0JFMSIgZD0iTTAgMGgxMnY0SDB6Ii8+PHBhdGggZmlsbD0iI0ZGRDUwMCIgZD0iTTAgNGgxMnYzSDB6Ii8+PHBhdGggZmlsbD0iI0UwQkMwMCIgZD0iTTAgN2gxMnYxSDB6Ii8+PC9zdmc+" style="padding-right: 0.3em;" alt="Slava Ukraini"> <a href="https://leafletjs.com" title="A JS library for interactive maps">Leaflet</a>');
+
+    let dialog = document.createElement("dialog");
+    dialog.setAttribute("class", "shortcuts-dialog");
+    dialog.setAttribute("autofocus", "");
+    dialog.onclick = function (e) {
+        e.stopPropagation();
+    };
+    dialog.innerHTML = `<b>${M.options.locale.kbdShortcuts} </b><button onclick='this.parentElement.close()'>X</button>` +
```
Comment (Malvoz) [inline src/mapml/control/AttributionControl.js:10]:
```suggestion
    dialog.innerHTML = `<b>${M.options.locale.kbdShortcuts} </b><button aria-label="Close" onclick='this.parentElement.close()'>X</button>` +
```

The close button is missing an accessible name. See [MDN: Interactive elements must be labeled](https://developer.mozilla.org/en-US/docs/Web/Accessibility/Understanding_WCAG/Text_labels_and_names#interactive_elements_must_be_labeled) for more info.

(I suppose "Close" should be translated as well at some point, per https://github.com/Maps4HTML/Web-Map-Custom-Element/issues/149.)

**Pull request merged**